pandas/core/reshape/reshape.py

```diff
@@ -1096,6 +1100,162 @@ def get_empty_frame(data) -> DataFrame:
         return DataFrame(dummy_mat, index=index, columns=dummy_cols)
 
 
+def from_dummies(
+    data: DataFrame,
+    sep: None | str = None,
+    implied_category: None | Hashable | dict[str, Hashable] = None,
+) -> DataFrame:
+    """
+    Create a categorical `DataFrame` from a `DataFrame` of dummy variables.
+
+    Inverts the operation performed by 'get_dummies'.
+
+    Parameters
+    ----------
+    data : DataFrame
+        Data which contains dummy-coded variables.
+    sep : str, default None
+        Separator used in the column names of the dummy categories they are
+        character indicating the separation of the categorical names from the prefixes.
+        For example, if your column names are 'prefix_A' and 'prefix_B',
+        you can strip the underscore by specifying sep='_'.
+    implied_category : None, Hashable or dict of Hashables, default None
```

Comment:
this is a non-obvious name. what has been considered here?

Reply:
It's meant to be the inverse of `drop_first` in `pd.get_dummies`

Not sure what a clearer name could be - `dropped_category` perhaps?

Reply:
I thought about using `dropped_category` but `implied_category` seemed more descriptive - `dropped` could be misunderstood as removed from the data (if looked at without the context of `drop_first` in `pd.get_dummies`).

To remove the ambiguity of what implies the `implied_category` we could use `base_category` instead. It should be more obivous in the sense of: it is the base category of a value if no other category is indicated.